apps/web/pages/refer-a-friend.tsx

```diff
@@ -0,0 +1,328 @@
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "@radix-ui/react-collapsible";
+import classNames from "classnames";
+import crypto from "crypto";
+import Head from "next/head";
+import { useState, useRef } from "react";
+import { Controller, useForm } from "react-hook-form";
+
+import { CAL_URL, WEBSITE_URL } from "@calcom/lib/constants";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import showToast from "@calcom/lib/notification";
+import { trpc } from "@calcom/trpc/react";
+import Button from "@calcom/ui/Button";
+import { Icon } from "@calcom/ui/Icon";
+import Shell from "@calcom/ui/Shell";
+import SkeletonLoader from "@calcom/ui/apps/SkeletonLoader";
+import { Form } from "@calcom/ui/form/fields";
+import { Input, Label } from "@calcom/ui/form/fields";
+
+import { defaultAvatarSrc } from "@lib/profile";
+
+import AvatarGroup from "@components/ui/AvatarGroup";
+import { AvatarSSR } from "@components/ui/AvatarSSR";
+
+interface Referrer {
+  id: number;
+  username: string | undefined;
+  referralPin: number | undefined;
+}
+
+interface Referees {
+  name: string | null;
+  username: string | null;
+  avatar: string | null;
+  email: string | undefined;
+  plan: string | undefined;
+}
+
+const ReferAFriend = () => {
+  const { t } = useLocale();
+  const { data, isLoading } = trpc.useQuery(["viewer.referrals.referralsQuery"], {
+    onSuccess: () => {
+      if (data) {
+        setReferralLink(
+          `${WEBSITE_URL}/signup/?referralCode=${data?.referrer?.username}${data?.referrer?.referralPin}`
+        );
+
+        setFreeReferees(data?.freeReferees);
+        setProReferees(data?.proReferees);
+      }
+    },
+  });
```

Comment:
Something that I can't figure out is sometimes the page loads but the referral link is empty. I know it involves state management but I can't wrap my head around it